include/IURLRequest.hpp

```diff
@@ -147,12 +147,14 @@ class IURLRequest
      * @param onSuccess Callback to be called when the request is successful.
      * @param onError Callback to be called when an error occurs.
      * @param fileName File name of output file.
+     * @param retryAttempts Number of retry attempts if the query fails.
      */
     virtual void get(
         const URL& url,
         std::function<void(const std::string&)> onSuccess,
         std::function<void(const std::string&)> onError = [](auto) {},
-        const std::string& fileName = "") = 0;
+        const std::string& fileName = "",
+        const unsigned int retryAttempts = 1) = 0;
```

Comment:
no retry as default value.

```suggestion
        const unsigned int retryAttempts = 0) = 0;
```